internal/pkg/service/stream/storage/statistics/repository/stats_allocation.go

```diff
@@ -0,0 +1,59 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/c2h5oh/datasize"
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
+)
+
+func (r *Repository) estimateSliceSizeOnSliceOpen() {
+	r.plugins.Collection().OnSliceOpen(func(ctx context.Context, now time.Time, file model.File, slice *model.Slice) error {
+		// The operation is performed immediately, outside the atomic operation!
+		// We need to update the value to the Slice entity before saving.
```

Comment:
To make it clear how it works :man_shrugging: it is clear also from the code, but just to be sure

Reply:
Ok, those questions are beter ;)

Reply:
> Why do we need to update the value before saving?

Because the entity must have the correct value set from the beginning so that the worker can create a file with reallocated space on the disk.

> Why can't it be part of the atomic operation?

Because it is only possible to modify the entity now, during the OnSliceOpen/OnSliceSave callback, later it is already added to the operations queue and cannot be modified.

Reply:
TODO: 

> later it is already added to the operations queue and cannot be modified.

Reply:
https://github.com/keboola/keboola-as-code/pull/1820/commits/79ed2e24429b0f46ab29d4f4391277ab65c46ee9